qcodes/instrument_drivers/Keysight/MSOS104A.py

```diff
@@ -0,0 +1,368 @@
+                                'Channel {} time series'.format(channel),),
+                         setpoint_units=('s',),
+                         docstring='raw trace from the scope',
+                         )
+        self._channel = channel
+        self._instrument = instrument
+
+
+    def set_sweep(self, start, stop, npts):
+        t = tuple(np.linspace(float(start), float(stop), num=npts))
+        self.setpoints = (t,)
+        self.shape = (npts,)
+
+
+    def get(self):
+        #self._instrument.write('DIGITIZE')
+        # oscil = self._instrument._parent.visa_handle
+        # the instrument should be used so self instead of oscil.
+        # ask and write are implemented, and they do exception handling
+        # and mocking
+        # also the parent is wrapped by the channel as a parent instrument
+        
+        # shorthand
+        oscil=self._instrument
+
+        # set up the instrument
+        # ---------------------------------------------------------------------
+
+        # check if requested number of points is less than 500 million
+
+        # realtime mode: only one trigger is used
+        instr._parent.acquire_mode('RTIMe')
+        
+        # acquire the data
+        # ---------------------------------------------------------------------
+
+        # digitize is the actual call for acquisition, blocks
+        oscil.write(':DIGitize CHANnel{}'.format(self._channel))
+
+        # switch display back on
+        oscil.write(':CHANnel1:DISPlay ON')
+
+        # transfer the data
+        # ---------------------------------------------------------------------
+
+        # switch the response header off for lower overhead
+        oscil.write(':SYSTem:HEADer OFF')
+        # select the channel from which to read
+        self._instrument._parent.data_source('CHAN{}'.format(self._channel))
+        # specifiy the data format in which to read
+        oscil.write(':WAVeform:FORMat WORD')
+        # streaming is only required for data > 1GB
+        oscil.write(':WAVeform:STReaming OFF')
+        # request the actual transfer
+        oscil.write(':WAVeform:DATA?')
+
+        visa_handle = oscil._parent.visa_handle
+        # first read the header
+        data = visa_handle.read_raw(2)
+
+        # then read the remaining data
+
+        # check if all data was received
+
+
+
+        npts = float(oscil.ask("WAV:POIN?"))
+        xorigin = float(oscil.ask(":WAVeform:XORigin?"))
+        yorigin = float(oscil.ask(":WAVeform:YORigin?"))
+        yinc = float(oscil.ask(":WAVeform:YINCrement?"))
+
+
+        srate = self.aquire_sample_rate.get_latest()
+
+        timestepsize = 1/srate
+        start = xorigin
+        stop = timestepsize*npts+xorigin
+        # what is the correct number of points?
+        npts = self.acquire_points.get_latest()
+
+        #
+        self.shape = (npts,)
+        self.set_sweep(start, stop, npts)
+        #TODO: fix above set when any of the above things change
```

Comment:
due to limitations in the dataset this cannot be set in the get command as the setpoints have already been written to fil

Reply:
Thank you, this is a good hint!